netbox/extras/views.py

```diff
@@ -1042,116 +1042,177 @@ def get_script_module(module, request):
     return get_object_or_404(ScriptModule.objects.restrict(request.user), file_path__regex=f"^{module}\\.")
 
 
-class ScriptView(ContentTypePermissionRequiredMixin, View):
+class BaseScriptView(ContentTypePermissionRequiredMixin, generic.ObjectView):
+    queryset = Script.objects.all()
+    script = None
+    script_class = None
+    jobs = None
 
     def get_required_permission(self):
         return 'extras.view_script'
 
-    def get(self, request, module, name):
+    def _init_vars(self, request):
+        if self.script.python_class:
+            self.script_class = self.script.python_class()
+        else:
+            self.script.delete_if_no_jobs()
+            messages.error(request, _("Script class has been deleted from module: ") + str(self.script.module))
+            if not self.script.id:
+                return redirect('extras:script_list')
+            else:
+                return redirect('extras:script_jobs', pk=self.script.id)
+
+        self.jobs = self.script.jobs.all()
+        return None
+
+    def get_script(self, request, pk):
+        self.script = get_object_or_404(Script.objects.all(), pk=pk)
+        return self._init_vars(request)
+
+    def get_script_by_module_name(self, request, module, name):
         module = get_script_module(module, request)
-        script = module.scripts[name]()
-        jobs = module.get_jobs(script.class_name)
-        form = script.as_form(initial=normalize_querydict(request.GET))
+        self.script = get_object_or_404(Script.objects.all(), module=module, name=name)
+        return self._init_vars(request)
+
+
+class ScriptView(BaseScriptView):
+
+    def get(self, request, pk):
+        if ret := self.get_script(request, pk):
+            return ret
```

Comment:
This is really confusing. It looks like we're using `get_script()` to return a redirect? I would expect it to return the script object.

Reply:
renamed to init_vars_or_redirect, hopefully that is more clear